modules/gapi/src/compiler/gcompiler.cpp

```diff
@@ -453,18 +453,32 @@ cv::GCompiled cv::gimpl::GCompiler::produceCompiled(GPtr &&pg)
     //     an execution plan for it (backend-specific execution)
     // ...before call to produceCompiled();
 
+    // Note: compile steps for python bridge :
+    // - First time python bridge calls compile with empty input meta (m_metas)
+    //   to collect input/output GTypesInfo to properly obtain python inputs
+    // - Once the python inputs are obtained, extract input meta and call compile once again
```

Comment:
I believe ideally this code shouldn't now anything about a python bridge and the hacks you put here.

It only increases coupling.

Reply:
Removed